File: EssayJoke/app/src/main/java/cn/xwj/essayjoke/BaseApplication.java
package cn.xwj.essayjoke;

import android.app.Application;
import android.util.Log;

import com.alipay.euler.andfix.patch.PatchManager;

import cn.xwj.crash.CrashExceptionHandler;
import cn.xwj.fixbug.FixDexManager;
import cn.xwj.util.Utils;
import cn.xwj.easy.E;

/**
 * Author: xw
 * Date: 2018-03-05 14:46:08
 * Description: BaseApplication <this is description>.
 */

public class BaseApplication extends Application {
    private static final String TAG = "BaseApplication";
    public static FixDexManager sFixDexManager;

    @Override
    public void onCreate() {
        super.onCreate();
        E.context(this);
        sFixDexManager = new FixDexManager(this);
        try {
            sFixDexManager.loadFixDex();
            Log.d(TAG, "sFixDexManager 加载完毕");
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}

File: KotlinMail/baselibrary/src/main/java/cn/xwj/baselibrary/common/BaseApplication.kt
package cn.xwj.baselibrary.common

import android.app.Application
import cn.xwj.baselibrary.di.component.AppComponent
import cn.xwj.baselibrary.di.component.DaggerAppComponent
import cn.xwj.baselibrary.utils.AppPreferences
import com.alibaba.android.arouter.launcher.ARouter

/**
 * Author: xw
 * Date: 2018-05-25 15:21:42
 * Description: BaseApplication: .
 */
open class BaseApplication : Application() {

    lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()
        initAppInjection()
        AppPreferences.instance.init(this)

        ARouter.openLog()
        ARouter.openDebug()
        ARouter.init(this)
    }

    private fun initAppInjection() {
        appComponent = DaggerAppComponent.builder()
                .context(this)
                .build()
    }

}